### PR TITLE
feature/AKD-107: Added Azure 'Point in Time Restore (PITR) Backup Retention Period' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/sqldatabases/pitrBackupEnabled.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    recommended_action: 'Ensure that an optimal backup retention period is set for Azure SQL databases.',
+    link: 'https://azure.microsoft.com/en-us/blog/azure-sql-database-point-in-time-restore/',
+    apis: ['servers:listSql', 'databases:listByServer', 'databases:shortTermRetentionPolicy'],
+    settings: {
+        ShortTemRetentionDays: {
+            name: 'Short Term Retention Period (Days)',
+            default: 7,
+            description: 'Number of days for which backups will be retained.',
+            regex: '^(3[0-5]|2[0-9]|1[0-9]|[1-9])$'
+        }
```
Comment (AkhtarAmir) [inline plugins/azure/sqldatabases/pitrBackupEnabled.js:18]:
```suggestion
        pitr_backup_retention_period: {
            name: 'Point in Time Restore Backup Retention Period',
            description: 'Desired number of days for which backups will be retained.',
            regex: '^(3[0-5]|2[0-9]|1[0-9]|[1-9])$',
            default: '7'
        }
```

File: plugins/azure/sqldatabases/pitrBackupEnabled.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    },
+
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+
+        const config = {
+            retentionDays: parseInt(settings.ShortTemRetentionDays || this.settings.ShortTemRetentionDays.default)
```
Comment (AkhtarAmir) [inline plugins/azure/sqldatabases/pitrBackupEnabled.js:27]:
```suggestion
            retentionDays: parseInt(settings.pitr_backup_retention_period || this.settings.pitr_backup_retention_period.default)
```

File: plugins/azure/sqldatabases/pitrBackupEnabled.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                helpers.addResult(results, 3,
+                    'Unable to query for SQL servers: ' + helpers.addError(servers), location);
+                return rcb();
+            }
+
+            if (!servers.data.length) {
+                helpers.addResult(results, 0, 'No SQL servers found', location);
+            }
+
+            for (const server of servers.data) {
```
Comment (AkhtarAmir) [inline plugins/azure/sqldatabases/pitrBackupEnabled.js:46]: Loop servers in async.each

File: plugins/azure/sqldatabases/pitrBackupEnabled.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
+var async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Point in Time Restore (PITR) Backup Retention Period',
```
Comment (AkhtarAmir) [inline plugins/azure/sqldatabases/pitrBackupEnabled.js:5]:
```suggestion
    title: 'Point in Time Restore (PITR) Backup Retention Period Configured',
```

File: plugins/azure/sqldatabases/pitrBackupEnabled.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    category: 'SQL Databases',
+    description: 'Ensures that Microsoft Azure SQL databases have a sufficient Point in Time Restore (PITR) backup retention period configured',
+    more_info: 'Point-in-time restore is a self-service capability, allowing customers to restore a Basic, Standard or Premium database from these backups to any point within the retention period.',
```
Comment (AkhtarAmir) [inline plugins/azure/sqldatabases/pitrBackupEnabled.js:8]:
```suggestion
    more_info: 'Point-in-time restore is a self-service capability, which enabled you to restore a database from backups to any point within the retention period. Point-in-time restore is useful in recovery scenarios, such as incidents caused by errors, incorrectly loaded data, or deletion of crucial data.',
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/sqldatabases/pitrBackupEnabled.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
+var async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Point in Time Restore (PITR) Backup Retention Period Configured',
```
Comment (giorod3) [inline plugins/azure/sqldatabases/pitrBackupEnabled.js:5]:
```suggestion
    title: 'Point in Time Restore Backup Retention',
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Review by giorod3 — APPROVED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**